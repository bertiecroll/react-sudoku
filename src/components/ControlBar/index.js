export { ControlBar } from './ControlBar'
export { default } from './ControlBar'
