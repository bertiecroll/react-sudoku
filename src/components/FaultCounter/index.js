export { FaultCounter } from './FaultCounter'
export { default } from './FaultCounter'
