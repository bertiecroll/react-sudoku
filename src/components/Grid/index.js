export { Grid } from './Grid'
export { GridRow } from './GridRow'
