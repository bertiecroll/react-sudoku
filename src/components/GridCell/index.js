export { GridCell } from './GridCell'
export { EmptyGridCell } from './EmptyGridCell'
export { CompleteGridCell } from './CompleteGridCell'
export { default } from './GridCell'
