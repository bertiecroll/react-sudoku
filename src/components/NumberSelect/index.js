export { NumberSelect } from './NumberSelect'
export { default } from './NumberSelect'
