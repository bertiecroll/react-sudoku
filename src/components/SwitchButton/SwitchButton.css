.SwitchButton {
  margin-top: 1em;
}

.SwitchButton button {
  width: 6em;
  padding: .5em;
  border: 1px solid #ccc;
  color: #67DAF9;
}

.SwitchButton button:focus {
  outline: none;
}

.SwitchButton .selected {
  border-color: darkblue;
  box-shadow: 0px 0px 3px 0px rgba(0, 0, 0, 0.2);
}

.SwitchButton button:first-child {
  background-color: #67DAF9;
  color: white;
  border-radius: 5px 0 0 5px;
}

.SwitchButton button:last-child {
  border-radius: 0 5px 5px 0;
}

.SwitchButton button:hover {
  cursor: pointer;
}
