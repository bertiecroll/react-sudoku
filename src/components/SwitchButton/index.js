export { SwitchButton } from './SwitchButton'
export { default } from './SwitchButton'
