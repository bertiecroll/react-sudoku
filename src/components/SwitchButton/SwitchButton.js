import React from 'react'
import { connect } from 'react-redux'

import './SwitchButton.css'
import { setCurrentMarker } from 'redux/actions/controlBar'
import { currentMarker, PEN_MARKER, PENCIL_MARKER } from 'redux/selectors/controlBar'

export function SwitchButton ({ currentMarker, setCurrentMarker }) {
  const handlePenButtonClick = () => setCurrentMarker(PEN_MARKER)
  const handlePencilButtonClick = () => setCurrentMarker(PENCIL_MARKER)

  return (
    <div className="SwitchButton">
      <button className={ currentMarker === PEN_MARKER ? 'selected' : '' } onClick={ handlePenButtonClick }>Pen</button>
      <button className={ currentMarker === PENCIL_MARKER ? 'selected' : '' } onClick={ handlePencilButtonClick }>Pencil</button>
    </div>
  )
}

const mapStateToProps = state => ({ currentMarker: currentMarker(state) })
const mapDispatchToProps = { setCurrentMarker }

export default connect(mapStateToProps, mapDispatchToProps)(SwitchButton)
