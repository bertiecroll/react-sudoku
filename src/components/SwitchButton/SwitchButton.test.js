import React from 'react'
import { shallow } from 'enzyme'
import { SwitchButton } from 'components/SwitchButton'
import { PEN_MARKER, PENCIL_MARKER } from 'redux/selectors/controlBar'

describe('SwitchButton', () => {
  const mockSetCurrentMarker = jest.fn()

  afterEach(() => {
    mockSetCurrentMarker.mockClear()
  })

  it('renders two button components and correct styles when props.currentMarker equals PEN_MARKER', () => {
    const wrapper = shallow(<SwitchButton setCurrentMarker={mockSetCurrentMarker} currentMarker={PEN_MARKER} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders two button components and correct styles when props.currentMarker equals PENCIL_MARKER', () => {
    const wrapper = shallow(<SwitchButton setCurrentMarker={mockSetCurrentMarker} currentMarker={PENCIL_MARKER} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('calls setCurrentMarker with PEN_MARKER when first button clicked', () => {
    const wrapper = shallow(<SwitchButton setCurrentMarker={mockSetCurrentMarker} currentMarker={PEN_MARKER} />)

    wrapper.find('button').first().simulate('click')

    expect(mockSetCurrentMarker).toHaveBeenCalledWith(PEN_MARKER)
  })

  it('calls setCurrentMarker with PENCIL_MARKER when second button clicked', () => {
    const wrapper = shallow(<SwitchButton setCurrentMarker={mockSetCurrentMarker} currentMarker={PENCIL_MARKER} />)

    wrapper.find('button').last().simulate('click')

    expect(mockSetCurrentMarker).toHaveBeenCalledWith(PENCIL_MARKER)
  })
})
