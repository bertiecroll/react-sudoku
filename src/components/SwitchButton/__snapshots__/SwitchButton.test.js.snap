// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwitchButton renders two button components and correct styles when props.currentMarker equals PEN_MARKER 1`] = `
<div
  className="SwitchButton"
>
  <button
    className="selected"
    onClick={[Function]}
  >
    Pen
  </button>
  <button
    className=""
    onClick={[Function]}
  >
    Pencil
  </button>
</div>
`;

exports[`SwitchButton renders two button components and correct styles when props.currentMarker equals PENCIL_MARKER 1`] = `
<div
  className="SwitchButton"
>
  <button
    className=""
    onClick={[Function]}
  >
    Pen
  </button>
  <button
    className="selected"
    onClick={[Function]}
  >
    Pencil
  </button>
</div>
`;
